src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx

```diff
@@ -133,32 +185,57 @@ export const PhrasalTemplatePhrase = ({
-              fields.filter(({ type }) => type === KEYWORDS.SENTENCE).length < 2;
-            const isOnlyResponseField =
-              field.type === KEYWORDS.ITEM_RESPONSE &&
-              fields.filter(({ type }) => type === KEYWORDS.ITEM_RESPONSE).length < 2;
-            const hasMinimumFields = fields.length > 2;
-
-            return (
-              <PhrasalTemplateField
-                canRemove={hasMinimumFields && !isOnlyResponseField && !isOnlySentenceField}
-                key={field.id}
-                name={`${phraseFieldsName}.${i}`}
-                onRemove={() => {
-                  remove(i);
-                }}
-                responseOptions={responseOptions}
-                type={field.type}
-                placeholder={fieldPlaceholders[i]}
-              />
-            );
-          })}
-        </StyledPhraseTemplateFieldSet>
+        <DragDropContext onDragEnd={handleDragEnd}>
+          <DndDroppable droppableId="phrasal-dnd-template" direction="vertical">
+            {(provided) => (
+              <Box {...provided.droppableProps} ref={provided.innerRef}>
+                <StyledPhraseTemplateFieldSet>
+                  {fields.map((field, i) => {
+                    const isOnlySentenceField =
+                      field.type === KEYWORDS.SENTENCE &&
+                      fields.filter(({ type }) => type === KEYWORDS.SENTENCE).length < 2;
+                    const isOnlyResponseField =
+                      field.type === KEYWORDS.ITEM_RESPONSE &&
+                      fields.filter(({ type }) => type === KEYWORDS.ITEM_RESPONSE).length < 2;
+                    const hasMinimumFields = fields.length > 2;
+
+                    return (
+                      <PhrasalTemplateField
+                        key={`draggable-item-${field.id}`}
+                        setAnchorPopUp={setAnchorPopUp}
+                        itemId={`draggable-item-${field.id}`}
+                        index={i}
+                        canRemove={hasMinimumFields && !isOnlyResponseField && !isOnlySentenceField}
+                        name={`${phraseFieldsName}.${i}`}
+                        onRemove={(e) => {
+                          setItemToDelete({
+                            item: i,
+                            shouldDelete: true,
+                            fieldName: field.type,
+                          });
 
+                          setAnchorPopUp(e.currentTarget);
+                        }}
+                        responseOptions={responseOptions}
+                        type={field.type}
+                        placeholder={fieldPlaceholders[i]}
+                      />
+                    );
+                  })}
+                </StyledPhraseTemplateFieldSet>
+              </Box>
+            )}
+          </DndDroppable>
+        </DragDropContext>
+        {itemToDelete.shouldDelete && !!anchorPopUp && (
+          <PhrasalTemplateDeleteItem
```

Comment:
Instead of utilizing a new custom component, there is an existing `Menu` component that can be utilized that will take care of transitions and a11y. See as an example [here](https://github.com/ChildMindInstitute/mindlogger-admin/blob/develop/src/modules/Builder/features/ActivityItems/ItemConfiguration/InputTypeItems/PhrasalTemplate/PhrasalTemplatePhrase/PhrasalTemplatePhrase.tsx#L174).